### ISSUE TITLE
Strange spikes in thermostat set temperature

### ISSUE THREAD

**Issue by glitter-ball** [proddy/EMS-ESP#489]
**Bug description**
Set room temperature from thermostat has weird spikes in it (see screenshot below). Some of these are linked to hot water firing - the pairs of spikes around 06:45, 08:45, 18:00, 20:54. Seems to generate two spikes - one when hot water heating starts and one when it stops. But, this doesn't explain spikes at 02:58 and 14:10 when hot water was not active. Nothing obvious in HA Logbook or History. But... Current Mode (derived from `ems-esp/thermostat_data1/mode`) keeps blipping from Auto to Manual about every 10 minutes (see screenshot below). Weirdness!.. Don't remember this with 1.9.5...

**Steps to reproduce**
Open set room temperature graph in HA, based on ems-esp/thermostat_data1/seltemp. Graph has weird temperature spikes at 63.5, 63, 55.5, 55 deg C - see screen shot.

**Expected behavior**
Graph to show typical set temperatures. In my case, 5 deg C overnight and when heating off. 18 or 18.5 C when heating on, without the spikes!

**Screenshots**
![image](https://user-images.githubusercontent.com/57917148/92797702-38bb2480-f3aa-11ea-8f40-cb9b092a8613.png)

Current Mode 00:00 to 03:00
![image](https://user-images.githubusercontent.com/57917148/92799347-9ef47700-f3ab-11ea-807c-e39b740c6389.png)

**Device information**
```
ems-esp:/$ show
EMS-ESP version 2.0.1b4

Boiler: Worcester Logamax Plus/GB192/Condens GC9000 (DeviceID:0x08 ProductID:208, Version:01.01)
  Hot tap water: off
  Central heating: off
  Warm Water activated: on
  Warm Water charging type: 3-way valve
  Warm Water circulation pump available: off
  Warm Water circulation pump freq: 0x3min
  Warm Water circulation active: off
  Warm Water comfort setting: Hot
  Warm Water disinfection temperature: 70°C
  Warm Water selected temperature: 60°C
  Warm Water set temperature: 60°C
  Warm Water current temperature (intern): 61.6°C
  Warm Water current temperature (extern): 61.6°C
  Warm water storage temperature (extern): 61.6°C
  Warm Water current tap water flow: 0.0l/min
  Warm Water # starts: 3445
  Warm Water active time: 25 days 2 hours 34 minutes
  Warm Water charging: on
  Warm Water disinfecting: off
  Selected flow temperature: 5°C
  Current flow temperature: 62.9°C
  Gas: off
  Boiler pump: off
  Fan: off
  Ignition: off
  Burner selected max power: 0%
  Burner current power: 0%
  Flame current: 0.0uA
  System service code:  (203)
  Heating temperature setting on the boiler: 75°C
  Boiler circuit pump modulation max power: 100%
  Boiler circuit pump modulation min power: 60%
  Boiler circuit pump delay time: 3min
  Boiler temp hysteresis on: -6°C
  Boiler temp hysteresis off: 4°C
  Boiler burner min period: 10min
  Boiler burner min power: 0%
  Boiler burner max power: 100%
  Outside temperature: 15.6°C
  Pump modulation: 0%
  Burner # starts: 12531
  Total burner operating time: 105 days 9 hours 34 minutes
  Total heat operating time: 80 days 7 hours 0 minutes
  Total UBA working time: 808 days 6 hours 16 minutes

Thermostat: RC300/RC310/Moduline 3000/CW400/Sense II (DeviceID:0x10, ProductID:158, Version:33.03)
  Clock: 21:12:46 10/09/2020
  Heating Circuit 1:
    Current room temperature: 21.2°C
    Setpoint room temperature: 18.5°C
    Mode: auto
    Mode Type: comfort
    Program is set to Summer mode
    Day temperature: 21.0°C
    Night temperature: 15.0°C
    Target flow temperature: 0°C

Dallas temperature sensors:
  ID: 28-BE2A-7791-0402-65, Temperature: 51.5°C

ems-esp:/$

```

**Additional context**
HA definition:
```
- platform: mqtt
  state_topic: 'ems-esp/thermostat_data1'
  name: 'Set Temperature'
  unit_of_measurement: '°C'
  icon: 'mdi:car-cruise-control'
  value_template: "{{ value_json.seltemp }}" 
```

**Comment by MichaelDvP** [comment]
I think that is the same issue we discussed on gitter concerning the mixer valve. Check if it is fixed in the the latest dev.

**Comment by MichaelDvP** [comment]
@glitter-ball have you seen these spikes again, or is it fixed?

**Comment by glitter-ball** [comment]
They are still there but, to confirm, I am still on 2.0.1b4. Do I need to upgrade? If so, can I do this from the GUI? I have values in the OTA Settings page but not sure how it works.

**Comment by glitter-ball** [comment]
I am also seeing duff data in my Dallas sensor in HA - spikes at -26240
![80378C52-6D6A-4BFF-9B98-04669014B891](https://user-images.githubusercontent.com/57917148/93017796-4eeffd00-f5c3-11ea-9f20-ce229637471b.jpeg)

This may or may not be related.

**Comment by glitter-ball** [comment]
I *think* I've upgraded - but not sure *8(.

I downloaded the ESP8266 .bin file from latest release assets and dragged and dropped that onto the firmware upgrade page in the GUI. It showed 100% upload immediately and that was that.  Waited it a bit and then restarted from telnet and the GUI now shows 2.0.1 instead of 2.0.1b4.

**Comment by glitter-ball** [comment]
OK. I think it is fixed. I re-flashed around 14:00 today. Prior to that you can see heating target temperature (olive) spikes every time boiler setpoint (blue) changes for hot water. At the last hot water re-heat, this didn't happen. I'll raise a separate issue for the large negative spikes on the Dallas sensor.

![image](https://user-images.githubusercontent.com/57917148/93023681-845b1180-f5e8-11ea-8982-7154be5ecb78.png)

**Comment by MichaelDvP** [comment]
> I think I've upgraded - but not sure *8(.

That's right, proddy has released the 2.0.1 today, there is also a identical 2.0.2b0 for download. 
Than i close this an you describe the dallas issue in a new issue.
